thirdparty/kyaml/runfn/runfn.go

```diff
@@ -110,6 +108,53 @@ func (r RunFns) Execute() error {
 	return r.runFunctions(nodes, output, fltrs)
 }
 
+func (r RunFns) getPipelineConfigFilterFn() (kio.LocalPackageSkipFileFunc, error) {
+	fnConfigPaths := sets.String{}
+	// TODO: eventually walking all sub-packages should be handled by
+	// pkg.FunctionConfigFilePaths
```

Comment:
We should implement function to gather function config paths (recursively) in the `pkg`. Extending given function or writing new function, I don't have strong opinion about that. Writing new function might makes more sense initially (logic can be shared eventually) but encapsulating file traversal and pkg/subpkg related code under `pkg` makes sense.
And then, implementing `kyaml.FileFilterFunc` on top of the `pkg` one.

Reply:
> Currently the subpkg is totally missed in `pkg`

Current one was implemented to support `fn render` use-case where subpkg support was not required. So it wasn't missed, it was designed as per requirement.

As I see it, you already have the logic implemented, so it's more about moving this logic to `pkg` as new function. Lets discuss this offline.